src/check/arbitrary/LetRecArbitrary.ts

```diff
@@ -75,8 +80,8 @@ function isLazyArbitrary(arb: Arbitrary<any> | undefined): arb is LazyArbitrary
  * @remarks Since 1.16.0
  * @public
  */
-export function letrec<T>(
-  builder: (tie: (key: string) => Arbitrary<unknown>) => { [K in keyof T]: Arbitrary<T[K]> }
+export function letrec<T, Cb extends TieFunction<T> = TieFunction<T>>(
```

Comment:
It seems that defining an overload helps to fix the issue reported by TS 3.2 🤔 

```ts
function letrec<T>(builder: (tie: (key: string) => Arbitrary<unknown>) => { [K in keyof T]: Arbitrary<T[K]>; }): { [K in keyof T]: Arbitrary<T[K]>; };
function letrec<T, Cb extends TieFunction<T> = TieFunction<T>>(builder: (tie: Cb) => { [K in keyof T]: Arbitrary<T[K]>; }): { [K in keyof T]: Arbitrary<T[K]>; };
```

Reply:
Actuallly does not seem to work well 😢 

I tried to check the typings by using:
```ts
expectType<{ a: fc.Arbitrary<number>; b: fc.Arbitrary<string>; c: fc.Arbitrary<string> }>()(
  fc.letrec<{ a: number; b: string; c: string }>((tie) => ({
    a: fc.nat(),
    b: fc.string(),
    c: tie('b'),
  })),
  'Invalid "letrec"'
);
```

But it fails to type it...